libvmi/driver/xen/xen_events.c

```diff
@@ -1410,3 +1417,147 @@ status_t xen_events_listen(vmi_instance_t vmi, uint32_t timeout)
 
     return vrc;
 }
+
+status_t xen_altp2m_get_domain_state ( vmi_instance_t vmi , bool *state )
```

Comment:
These functions should be in a separate file in the Xen folder, say altp2m.c, as these don't belong to the events system.